internal/auth/query.go

```diff
@@ -0,0 +1,844 @@
+     where public_id in (select public_id from auth_methods)
+),
+password as (
+    select *
+      from auth_password_method_with_is_primary
+     where public_id in (select public_id from auth_methods)
+),
+final as (
+    select public_id,
+           scope_id,
+           case when ldap.is_primary_auth_method is true then
+             'ldap'
+           else null end
+           as primary_auth_method_subtype,
+           name,
+           description,
+           create_time,
+           update_time,
+           version,
+           state,
+           start_tls,
+           insecure_tls,
+           discover_dn,
+           anon_group_search,
+           upn_domain,
+           enable_groups,
+           use_token_groups,
+           maximum_page_size,
+           urls,
+           certs,
+           account_attribute_map,
+           user_dn,
+           user_attr,
+           user_filter,
+           group_dn,
+           group_attr,
+           group_filter,
+           client_certificate_key,
+           client_certificate_key_hmac,
+           client_certificate_key_id,
+           client_certificate_cert,
+           bind_dn,
+           bind_password,
+           bind_password_hmac,
+           bind_password_key_id,
+           dereference_aliases,
+           null as disable_discovered_config_validation, -- Add to make union uniform
+           null as api_url,
+           null as issuer, 
+           null as client_id, 
+           null as client_secret, 
+           null as client_secret_hmac, 
+           null as key_id, 
+           null as max_age,
+           null as algs,
+           null as auds,
+           null as certs,
+           null as claims_scopes,
+           null as prompts,
+           null as account_claim_maps,
+           null as password_conf_id,          -- Add to make union uniform
+           null::integer as min_login_name_length,  
+           null::integer as min_password_length,
+           'ldap' as subtype
+      from ldap
+     union
+    select public_id,
+           scope_id,
+           case when oidc.is_primary_auth_method is true then
+             'oidc'
+           else null end
+           as primary_auth_method_subtype,
+           name,
+           description,
+           create_time,
+           update_time,
+           version,
+           state,
+           null as start_tls,                    -- Add to make union uniform
+           null as insecure_tls,                 -- Add to make union uniform
+           null as discover_dn,                  -- Add to make union uniform
+           null as anon_group_search,            -- Add to make union uniform
+           null as upn_domain,                   -- Add to make union uniform
+           null as enable_groups,                -- Add to make union uniform
+           null as use_token_groups,             -- Add to make union uniform
+           null as maximum_page_size,            -- Add to make union uniform
```

Comment:
I see you got tired of adding these. IMO either do it for every line or just do it for the first in the sequence consistently 😂

Reply:
I swear I intended to come back and finish them, once I had figured out exactly which ones needed to be added 😂